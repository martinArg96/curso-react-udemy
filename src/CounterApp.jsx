import { useState } from 'react';
import PropTypes from 'prop-types';
//los hooks son funciones


export const  CounterApp = ({value}) => {

    console.log('renderisado')
    
    const [ counter, setCounter ] = useState (   value   );//valor incial es value

    
    const handleAdd = () => {
        // console.log('+1')
        // value = 1000;
        // console.log(value)
        setCounter( counter +  1  )
    }
    const handleRest = () => {
        setCounter( counter - 1)
    }
    const Reset = () => {
        setCounter( value )
    }

    return(
        <>
        <h1> Counter App</h1>
        <h2>    {   counter   }   </h2>
        <button onClick={ /*(event) => handleAdd(event) */ handleAdd } >
            +1
        </button>
        <button onClick={ /*(event) => handleAdd(event) */ handleRest } >
            -1
        </button>
        <button onClick={ /*(event) => handleAdd(event) */ Reset } >
            reset
        </button>

        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

CounterApp.defaultProps = {
    value: 1
}