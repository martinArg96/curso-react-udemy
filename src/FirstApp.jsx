import PropTypes from 'prop-types';//hay que instalar manualmente npm install prop-types

const newMessage = {
    message: 'hola mundo',
    title: 'martin',
    edad: 34
};
const martin = {
    nombre: 'martin',
    apellido: 'drab',
    edad: 25,
    casado: false,
};

// function getMessage(){
//     return martin
// }

    export const FirstApp = (   {title , subtitle }   ) => {
//    console.log(props)
    // if(  !title   ){
    //     throw new Error('el titulito no existe');
    // }
    return(

        //siempre hay que llevar un solo elemento padre para eso usamos <>....</> si son varias cosas
        <> 
            
          {/* ?\  <h1>    {  JSON.stringify (  getMessage() )   }  </h1> */}
            <code>  {   title   }   </code>
            <p> {   subtitle    }  </p>
        </>
    
    ) 
}
//IMPORTANTE NO SE  PUEDE PONER OBJETOS DENTRO DEL RETURN

//creamos objeto para la protypes. antes installar npm install proptypes e importar al principio del documento
FirstApp.propTypes = {
    title: PropTypes.string.isRequired, //obligamos que sea tipo string y que sea requerido
    subtitle: PropTypes.number.isRequired,
}
FirstApp.defaultProps = {
    title: 'no hay titulo',
    subtitle: 'no hay subtitulo'
}