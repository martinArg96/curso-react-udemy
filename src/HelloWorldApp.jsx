// export function App() {
//     return (<h1> Hola mundo !!!!</h1>)
// }

export const HelloWorldApp = () => {

    return(
        <h1> Hello world App</h1>
    );
}

