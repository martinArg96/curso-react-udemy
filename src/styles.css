html,
body{
    background-color: rgba(31, 29, 29, 0.852);
    color: bisque;
    font-family: sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}

button{
    padding: 15px;
    font-size: 1.5rem;
}