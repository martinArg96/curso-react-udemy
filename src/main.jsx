//Primero importo React
import React from 'react' ;

//una herramienta para renderizar
import ReactDOM from 'react-dom/client';
import {HelloWorldApp} from './HelloWorldApp'; 
import {FirstApp} from './FirstApp';
import './styles.css';
import { CounterApp } from './CounterApp';


//



//empezamos con un componente... funcional component... porque es una funcion
//en general no esta dentro del main
/*
function App() {
    return (<h1> Hola mundo !!!!</h1>)
}
*/
//comento la function app y la llevo a helloworldapp.js(al final)

//importo direcatmnte
//import {App} from './HelloWorldApp'


//Renderizando cen el DOM 
// .....Document Object Model: Estandar para la presentacion de un documento HTML. 
//......Define la estrcutura y al forma de acceder y manipular objetos
/*
ReactDOM.createRoot(    document.getElementById('root')).render(
    <React.StrictMode>
        <HelloWorldApp/>
    </React.StrictMode>);
*/
ReactDOM.createRoot(    document.getElementById('root')).render(
    <React.StrictMode>
        {/* <CounterApp value={5} /> */}
        <FirstApp/>
    </React.StrictMode>);


