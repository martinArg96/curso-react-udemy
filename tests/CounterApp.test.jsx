import { CounterApp } from "../src/CounterApp"
import {render,screen,fireEvent} from '@testing-library/react'
import { stringify } from "ts-jest"

describe('////--------Pruebas en Counter App------------///', () => {
    
    const initialValue = 10
    
    test('Debe de bhacer march con el snapshot', () => { 
        
        const {container} = render(<CounterApp/>)
        expect( container ).toMatchSnapshot()
        
    
    
    })
    
    test('Debe de mostrar el valor incial de 100', () => { 
        
        render(<CounterApp value={100}/>)
        console.log(( screen.getByRole('heading',{level:2 }).innerHTML))
        
        expect( screen.getByText(100)).toBeTruthy
    })

    test('debe de incrementar con el boton +1', () => {
        
        render(<CounterApp value={initialValue}/> ) 

        fireEvent.click(    screen.getByText('+1'))
        console.log(stringify(initialValue+1))
        expect(screen.getByText(stringify(initialValue+1))).toBeTruthy


    
    
    })
    
    test('debe de decrementar con el boton -1', () => {
        
        render(<CounterApp value={initialValue}/> ) 

        fireEvent.click(    screen.getByText('-1'))
        screen.debug()
        fireEvent.click(    screen.getByText('-1'))
        screen.debug()
        fireEvent.click(    screen.getByText('-1'))
        screen.debug()
        console.log(stringify(initialValue-1))
        expect(screen.getByText(stringify(initialValue-3))).toBeTruthy

    })
    

    test('debe de funcionar el boton de reset', () => { 
        render(<CounterApp value={400}/> ) 

        fireEvent.click(    screen.getByText('+1'))
        fireEvent.click(    screen.getByText('+1'))
        fireEvent.click(    screen.getByText('+1'))
        fireEvent.click(    screen.getByText('+1'))
        screen.debug()
        fireEvent.click( screen.getByRole('button',{name: 'reset'}))

        screen.debug()

        expect(screen.getByText(400)).toBeTruthy
    })
    
    
    
    
    
    
})