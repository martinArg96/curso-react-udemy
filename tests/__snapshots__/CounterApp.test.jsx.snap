// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`////--------Pruebas en Counter App------------/// Debe de bhacer march con el snapshot 1`] = `
<div>
  <h1>
     Counter App
  </h1>
  <h2>
        
    1
       
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button
    aria-level="btn-reset"
  >
    reset
  </button>
</div>
`;
